Source/WebCore/rendering/RenderLayerModelObject.cpp

```diff
@@ -133,8 +133,8 @@ void RenderLayerModelObject::styleWillChange(StyleDifference diff, const RenderS
 
 void RenderLayerModelObject::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle)
 {
-    RenderElement::styleDidChange(diff, oldStyle);
     updateFromStyle();
+    RenderElement::styleDidChange(diff, oldStyle);
 
```

Comment:
I am surprised no code relies on this odd state where style is already set on the renderer but caches haven't been populated yet. I guess if something starts failing that should probably go to styleWillChange anyway.
Also why isn't it in RenderElement! RenderLayerModelObject sounds like a random place for a function like "update from style" -though I assume it is driven by actual use cases.